instructor/client_vertexai.py

```diff
@@ -15,7 +15,9 @@ def _create_gemini_json_schema(model: BaseModel):
     gemini_schema: dict[Any, Any] = {
         "type": schema_without_refs["type"],
         "properties": schema_without_refs["properties"],
-        "required": schema_without_refs["required"],
+        "required": schema_without_refs["required"]
+        if "required" in schema_without_refs
+        else [],  # TODO: Temporary Fix for Iterables which throw an error when their tasks field is specified in the required field
```

Comment:
When the tests from `test_stream` were being ran, we were getting an error that schema_without_refs['required'] had a key error.

I realised it's because the schema we generate is of type `IterableBaseModel`. For the specific test case we have this schema generated

```json
{
    "$defs": {
        "UserExtract": {
            "properties": {
                "name": {"title": "Name", "type": "string"},
                "age": {"title": "Age", "type": "integer"},
            },
            "required": ["name", "age"],
            "title": "UserExtract",
            "type": "object",
        }
    },
    "description": "Correct segmentation of `UserExtract` tasks",
    "properties": {
        "tasks": {
            "description": "Correctly segmented list of `UserExtract` tasks",
            "items": {
                "properties": {
                    "name": {"title": "Name", "type": "string"},
                    "age": {"title": "Age", "type": "integer"},
                },
                "required": ["name", "age"],
                "title": "UserExtract",
                "type": "object",
            },
            "title": "Tasks",
            "type": "array",
        }
    },
    "title": "IterableUserExtract",
    "type": "object",
}
```

Required doesn't exist on the schema we've defined here but it does if say we use a normal response object.

```
class UserDetails(BaseModel):
        name: str
        age: int
```

This in turn generates the schema below where we have a `required` property. 

```
{
    "properties": {
        "name": {"title": "Name", "type": "string"},
        "age": {"title": "Age", "type": "integer"},
    },
    "required": ["name", "age"],
    "title": "UserDetails",
    "type": "object",
}
```

I think the easiest way is just to check if the key exists, and return an empty list otherwise